modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/sequence/HeadersSideBar.scala

```diff
@@ -1,24 +1,44 @@
 package edu.gemini.seqexec.web.client.components.sequence
 
 import diode.react.ModelProxy
-
 import edu.gemini.seqexec.web.client.components.DropdownMenu
 import edu.gemini.seqexec.web.client.semanticui.elements.label.Label
-import edu.gemini.seqexec.web.client.semanticui.elements.input.Input
+import edu.gemini.seqexec.web.client.semanticui.elements.input.InputEV
 import edu.gemini.seqexec.web.client.model._
-
-import japgolly.scalajs.react.{ReactComponentB, Callback}
+import japgolly.scalajs.react.{BackendScope, Callback, ReactComponentB, ReactComponentU, TopNode}
+import japgolly.scalajs.react.extra.{ExternalVar, TimerSupport}
 import japgolly.scalajs.react.vdom.prefix_<^._
+import org.scalajs.dom.html.Div
+
+import scalaz.syntax.equal._
+import scalaz.std.string._
+import scalaz.std.option._
+import scala.concurrent.duration._
+import scalaz.Equal
 
 /**
   * Display to show headers per sequence
   */
 object HeadersSideBar {
   case class Props(operator: ModelProxy[Option[String]], status: ModelProxy[ClientStatus])
 
-  val component = ReactComponentB[Props]("HeadersSideBar")
-    .stateless
-    .render_P(p =>
+  case class State(currentText: Option[String])
+
+  object State {
+    implicit val equals: Equal[State] = Equal.equalA[State]
+  }
+
+  class Backend(val $: BackendScope[Props, State]) extends TimerSupport {
+    def updateState(value: String): Callback =
+      $.modState(_.copy(currentText = Some(value)))
+
+    def submitIfChanged: Callback =
+      ($.state zip $.props) >>= {
+        case (s, p) => Callback.when(s.currentText =/= p.operator())(Callback.empty) // We are not submitting until the backend properly update this property
+      }
+
+    def render(p: Props, s: State): ReactTagOf[Div] = {
+      val operatorEV = ExternalVar(s.currentText.getOrElse(""))(updateState)
```

Comment:
They are essentially safe vars that can be used to communicate across components. It is a `scalajs-react` facility that in this case lets state flow from the component `InputEV` to its parent` HeadersSideBar`, complementing the strict top-down data flow used by `diode`

It was the only way I found to this without inducing UI glitches, in particular, the input element would loose focus when typing doing the top-down approach